plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,202 @@
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/classes/xr_server.hpp>
+
+#include <godot_cpp/variant/utility_functions.hpp>
+
+OpenXRFbSpaceWarpExtensionWrapper *OpenXRFbSpaceWarpExtensionWrapper::singleton = nullptr;
+
+OpenXRFbSpaceWarpExtensionWrapper *OpenXRFbSpaceWarpExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbSpaceWarpExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbSpaceWarpExtensionWrapper::OpenXRFbSpaceWarpExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbSpaceWarpExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_SPACE_WARP_EXTENSION_NAME] = &fb_space_warp_ext;
+	singleton = this;
+}
+
+OpenXRFbSpaceWarpExtensionWrapper::~OpenXRFbSpaceWarpExtensionWrapper() {
+	singleton = nullptr;
+}
+
+godot::Dictionary OpenXRFbSpaceWarpExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	if (fb_space_warp_ext) {
+		system_space_warp_properties.next = p_next_pointer;
+		return reinterpret_cast<uint64_t>(&system_space_warp_properties);
+	} else {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_projection_views_and_get_next_pointer(int p_view_index, void *p_next_pointer) {
+	if (enabled) {
+		return reinterpret_cast<uint64_t>(&space_warp_info[p_view_index]);
+	} else {
+		return 0;
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_created(uint64_t p_instance) {
+	if (fb_space_warp_ext) {
+		get_openxr_api()->register_projection_views_extension(this);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_destroyed() {
+	if (fb_space_warp_ext) {
+		get_openxr_api()->unregister_projection_views_extension(this);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_state_ready() {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+	int width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+
+	motion_vector_swapchain_info = get_openxr_api()->openxr_swap_chain_info_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT, 0x881A, width, height, 1, view_count);
+	motion_vector_depth_swapchain_info = get_openxr_api()->openxr_swap_chain_info_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT, 0x88F0, width, height, 1, view_count);
```

Comment:
```suggestion
	motion_vector_swapchain_info = get_openxr_api()->openxr_swap_chain_info_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT, 0x881A, width, height, 1, view_count);
	motion_vector_depth_swapchain_info = get_openxr_api()->openxr_swap_chain_info_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT, 0x88F0, width, height, 1, view_count);
```

We should use `#define`s for the OpenGL constants here (ie `0x881A` and `0x88F0`).

Also, related to [this comment](https://github.com/godotengine/godot/pull/97151#pullrequestreview-2395820410) on the Godot PR, we should be checking both that (1) OpenXR is using OpenGL and (2) that these swapchain formats are supported by the runtime. But that'll need the new APIs discussed there first.

Reply:
Makes sense, I've converted these to defines and will work on the other functionality 👍